pixsfm/pyceres/problem.cc

```diff
@@ -93,6 +100,47 @@ class CostFunctionWrapper : public ceres::CostFunction {
   ceres::CostFunction* cost_function_;
 };
 
+#if CERES_USE_MANIFOLD
+class LocalParameterizationWrapper : public ceres::Manifold {
+ public:
+  explicit LocalParameterizationWrapper(ceres::Manifold* real_parameterization)
+      : parameterization_(real_parameterization) {}
+  virtual ~LocalParameterizationWrapper() {}
+
+  bool Plus(const double* x, const double* delta,
+            double* x_plus_delta) const override {
+    return parameterization_->Plus(x, delta, x_plus_delta);
+  }
+
+  bool PlusJacobian(const double* x, double* jacobian) const override {
+    return parameterization_->PlusJacobian(x, jacobian);
+  }
+
+  bool RightMultiplyByPlusJacobian(const double* x, const int num_rows,
+                          const double* global_matrix,
+                          double* local_matrix) const override {
+    return parameterization_->RightMultiplyByPlusJacobian(
+        x, num_rows, global_matrix, local_matrix);
+  }
+
+  bool Minus(const double* y, const double* x,
+             double* y_minus_x) const override {
+    return parameterization_->Minus(y, x, y_minus_x);
+  }
+
+  bool MinusJacobian(const double* x, double* jacobian) const override {
+    return parameterization_->MinusJacobian(x, jacobian);
+  }
+
+  int AmbientSize() const override { return parameterization_->AmbientSize(); }
+
+  // Size of delta.
+  int TangentSize() const override { return parameterization_->TangentSize(); }
+
+ private:
+  ceres::Manifold* parameterization_;
+};
+#else
 class LocalParameterizationWrapper : public ceres::LocalParameterization {
```

Comment:
@Phil26AT Why do we need this wrapper? Why isn't `PyLocalParameterization` sufficient?